certified-connectors/DocuSignDemo/script.csx

```diff
@@ -1690,21 +1673,29 @@ public class Script : ScriptBase
 
             if (phoneNumber.ToString().Equals(signerPhoneNumber))
             {
-              newSigner["recipientId"] = signer["recipientId"];
-              newSigner["routingOrder"] = signer["routingOrder"];
-              newSigner["roleName"] = signer["roleName"];
-              newSigner["name"] = signer["name"];
-              newSigner["email"] = signer["email"];
-              newSigner["recipientType"] = signer["recipientType"];
-              newSigner["verificationType"] = signer["verificationType"];
-              newSigner["recipientIdGuid"] = signer["recipientIdGuid"];
+              newSigner["signer"] = signer;
               break;
             }
           } 
         }
       }
 
-      response.Content = new StringContent(newSigner.ToString(), Encoding.UTF8, "application/json");
+      if ((recipientEmailId == null) && (query.Get("phoneNumber") == null))
+      {
+        throw new ConnectorException(HttpStatusCode.BadRequest, "ValidationFailure: Please fill either Recipient Email or Phone Number to retrieve Recipient information");
+      } 
+      else 
+      {
+        newBody["recipientId"] = newSigner["signer"]["recipientId"];
```

Comment:
if you use recommendation mentioned above, you can leave the variable name as signer and also avoid double referencing.
 e.g : newBody["recipientId"] = signer["recipientId"];

Reply:
Cannot implement this due to the reason above.

Reply:
with above method it simply becomes like following, thus avoiding extra level of indirection
` newBody["recipientId"] = matchingSigner["recipientId"];`